MANIFEST.in

```diff
@@ -1,31 +1,38 @@
-include MANIFEST.in
+# Include source files in SDist
 include CMakeLists.txt
+include *.bzl
+include *.bazel
+include WORKSPACE
+include requirements.txt
+include version.txt
+include .gitmodules
+graft android
+graft aten
+graft binaries
+graft c10
+graft caffe2
+graft cmake
+graft functorch
+graft ios
+graft third_party
+graft tools
+graft torch
+graft torchgen
+
+# Misc files and directories in SDist
 include CITATION.cff
 include LICENSE
 include NOTICE
-include .gitmodules
-include build_variables.bzl
 include mypy.ini
-include requirements.txt
-include ufunc_defs.bzl
-include version.txt
-recursive-include android *.*
-recursive-include aten *.*
-recursive-include binaries *.*
-recursive-include c10 *.*
-recursive-include caffe2 *.*
-recursive-include cmake *.*
-recursive-include torch *.*
-recursive-include tools *.*
-recursive-include test *.*
-recursive-include docs *.*
-recursive-include ios *.*
-recursive-include third_party *
-recursive-include test *.*
-recursive-include benchmarks *.*
-recursive-include scripts *.*
-recursive-include mypy_plugins *.*
-recursive-include modules *.*
-recursive-include functorch *.*
+graft benchmarks
+graft docs
+graft mypy_plugins
+graft scripts
+
+# Include test suites in SDist
+graft test
+
+# Prune compiled files
 prune */__pycache__
-global-exclude *.o *.so *.dylib *.a .git *.pyc *.swp
+global-exclude *.o *.obj *.so *.a *.dylib *.pxd *.dll *.lib *.py[cod]
+global-exclude .git *~ *.swp
```

Comment:
I don't think we can do this as this will break when we try check out the nccl submodule while building CUDA... or is .gitmodules really all you need?